### DISCUSSION TITLE
.NET Core SDK container images now contain the Visual Studio 2019 toolset

### DISCUSSION THREAD

**Discussion by MichaelSimons** [dotnet/dotnet-docker#3655] [Announcements]
# .NET Core SDK container images now contain the Visual Studio 2019 toolset

.NET Core SDK container images now include Visual Studio 2019 tools. For most scenarios, this change will not be observable, however, there may be some breaks that will be documented in this issue. We did not add *additional* tools to the SDK, but updated existing tools to match the same tools in Visual Studio 2019. Previously, the tools in SDK matched the same tools in Visual Studio 2017.

We will only produce .NET Core SDK images that include Visual Studio 2019 tools. We will not produce another set of .NET Core SDK images that includes Visual Studio 2017 tools. There is a strong goal that the tools remain compatible over releases.

## Details

[Visual Studio 2019](https://devblogs.microsoft.com/visualstudio/visual-studio-2019-code-faster-work-smarter-create-the-future/) was released on April 2, 2019. We updated the .NET Core SDK container images, on the same day, to include a new version of the .NET Core SDK that contains tools that match the same tools in Visual Studio 2019.

These tools include:

* MSBuild
* NuGet
* C# compiler
* F# compiler
* VB compiler
* VS Test

If you encounter a breaking change or other issues and need to quickly unblock yourself, you can reference the previous version of the image by using its full version tag:

* 2.1: `mcr.microsoft.com/dotnet/core/sdk:2.1.505`
* 2.2: `mcr.microsoft.com/dotnet/core/sdk:2.2.105`

Details:

* [dotnet/dotnet-docker #1029](https://github.com/dotnet/dotnet-docker/pull/1029)
* [2.1 SDK release notes](https://github.com/dotnet/core/blob/master/release-notes/2.1/2.1.602-SDK/2.1.602-SDK.md)
* [2.2 SDK release notes](https://github.com/dotnet/core/blob/master/release-notes/2.2/2.2.202-SDK/2.2.202-SDK.md)

## Known Breaking Changes for SDK images (VS 2017 to VS 2019 upgrade)

The following issues and breaks have been discovered.
 
### Test Log File Name Change

A change was introduced to the `dotnet test` command when a user-defined log file name is specified.  In the previous version the resulting log file would be named exactly as specified.  In the current version, the log file is named with a timestamp suffix automatically added to the user-defined name.  

Details:
* [Microsoft/vstest #1603](https://github.com/Microsoft/vstest/issues/1603) 
* [Microsoft/vstest#1876](https://github.com/Microsoft/vstest/issues/1876)

**Comment by niklaslundberg** [comment]
We are targeting .NET Core 2.2 and use https://www.nuget.org/packages/dotnet-sonarscanner/. Sonar scanner currently only supports VS2017 tool chain. Updating to the latest Docker SDK image 2.2.203 does not work and there is no image for 2.2.106. Thus we cannot use the latest 2.2.4 runtime in any way.

**Comment by nguerrera** [comment]
Seems that latest version of sonarscanner works with 2019 toolchain: https://github.com/SonarSource/sonar-scanner-msbuild/issues/719#issuecomment-481699634
